core/api.go

```diff
@@ -250,14 +253,17 @@ func (a *API) authnHandler(handler http.Handler) http.Handler {
 		req, err := auth.Authenticate(req)
 		if err != nil {
 			errorFormatter.Write(req.Context(), rw, errNotAuthenticated)
+			return
 		}
 		handler.ServeHTTP(rw, req)
 	})
 }
 
-func authzHandler(handler http.Handler) http.Handler {
+func (a *API) authzHandler(handler http.Handler) http.Handler {
 	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-		if !authz.Authorized(req.Context()) {
+		auth := authz.NewAuthorizer(a.raftDB, grantPrefix, policyByRoute)
```

Comment:
Since the authorizer is basically stateless, this would work fine, but still I think it would be better to define this outside the closure.